src/components/Dropdown.vue

```diff
@@ -0,0 +1,152 @@
+<script lang="ts">
+import { computed, defineComponent, PropType, ref } from "vue";
+import DropdownContent from "./DropdownContent.vue";
+
+export type Options = string[];
+
+type Props = {
+  options: Options;
+  selectedOption: string;
+  label: string;
+  width: string;
+};
+
+export default defineComponent({
+  components: { DropdownContent },
+  props: {
+    options: {
+      type: Object as PropType<Options>,
+      required: true,
+    },
+    selectedOption: {
+      type: String,
+      required: true,
+    },
+    label: {
+      type: String,
+      default: "",
+    },
+    width: {
+      type: String,
+      default: "8.8rem",
+    },
+  },
+  emits: ["update:selectedOption"],
+  setup(props: Props, { emit }) {
+    const isOptionsShown = ref(false);
+
+    const isDefault = computed(() => {
+      return ["", "指定なし"].includes(props.selectedOption);
+    });
+
+    const hasLabel = computed(() => {
+      return props.label !== "";
+    });
+
+    const isSelected = (value: string) => {
+      return value === props.selectedOption;
+    };
+
+    const toggleShown = () => {
+      isOptionsShown.value = !isOptionsShown.value;
+    };
+
+    const closeOptions = () => {
+      isOptionsShown.value = false;
+    };
+
+    const emitSelectedEvent = (option: string) => {
+      emit("update:selectedOption", option);
+    };
+
+    return {
+      isOptionsShown,
+      isDefault,
+      isSelected,
+      hasLabel,
+      toggleShown,
+      closeOptions,
+      emitSelectedEvent,
+    };
+  },
+});
+</script>
+
+<template>
+  <div class="dropdown">
+    <div v-if="hasLabel" class="dropdown__label">{{ label }}</div>
+    <div
+      @click="toggleShown"
+      v-click-away="closeOptions"
+      class="dropdown__box"
+      :style="{ width: width }"
+    >
+      <div v-if="isDefault" class="dropdown__box__text--unselected">
+        指定なし
+      </div>
+      <div v-else>{{ selectedOption }}</div>
+      <div
+        :class="{
+          'material-icons': true,
+          dropdown__box__icon: true,
+          '--turned': isOptionsShown,
+        }"
+      >
+        expand_more
+      </div>
+    </div>
+    <transition name="spread-down">
+      <div v-show="isOptionsShown" class="dropdown__options">
+        <DropdownContent
+          v-for="option in options"
+          :key="option"
+          :value="option"
+          :isSelected="isSelected(option)"
+          :width="width"
+          @click="[emitSelectedEvent(option), toggleShown()]"
+        ></DropdownContent>
+      </div>
+    </transition>
+  </div>
+</template>
+
+<style scoped lang="scss">
+@import "../scss/main.scss";
```

Comment:
絶対パスにしようかなって思ってる（一部未完）
src通るやつは基本的に絶対パスにするように統一したほうがわかりやすいかな、ってなった
```suggestion
@import "~/scss/main.scss";
```